lib/integrations/awsoidc/credprovider/integration_config_provider_test.go

```diff
@@ -0,0 +1,161 @@
+package credprovider
+
+import (
+	"context"
+	"crypto"
+	"testing"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/keys"
+	"github.com/gravitational/teleport/lib/auth/testauthority"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	"github.com/gravitational/teleport/lib/integrations/awsoidc"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local"
+)
+
+const (
+	integrationName = "test-integration1"
+	awsRegion       = " eu-central-1"
+	testUser        = "test-user"
+)
+
+func TestCreateAWSConfigForIntegration(t *testing.T) {
+	deps := newDepsMock(t)
+	ctx := context.Background()
+
+	t.Run("should auto init credentials during retrieve call", func(t *testing.T) {
+		config, err := CreateAWSConfigForIntegration(ctx, Config{
+			Region:                awsRegion,
+			IntegrationName:       integrationName,
+			IntegrationGetter:     deps,
+			AWSOIDCTokenGenerator: deps,
+			STSClient:             &fakeSTSClient{clock: clockwork.NewFakeClock()},
+		})
+		require.NoError(t, err)
+
+		creds, err := config.Credentials.Retrieve(ctx)
+		require.NoError(t, err)
+		require.NotEmpty(t, creds.SecretAccessKey)
+	})
+
+	t.Run("should init creds before retrieve call", func(t *testing.T) {
+		config, err := CreateAWSConfigForIntegration(ctx, Config{
+			Region:                awsRegion,
+			IntegrationName:       integrationName,
+			IntegrationGetter:     deps,
+			AWSOIDCTokenGenerator: deps,
+			STSClient:             &fakeSTSClient{clock: clockwork.NewFakeClock()},
+		}, WithWaitForFirstInit(true))
+		require.NoError(t, err)
+
+		creds, err := config.Credentials.Retrieve(ctx)
+		require.NoError(t, err)
+		require.NotEmpty(t, creds.SecretAccessKey)
```

Comment:
since we are testing this to be generated before retrieve call...shouldn't `require.NotEmpty(t, creds.SecretAccessKey)` be checked before `creds, err := config.Credentials.Retrieve(ctx)`?

Reply:
yea, let check that. Added that  AssumeRoleWithWebIdentity call count assertion check: 
https://github.com/gravitational/teleport/pull/48812/files#diff-124f07f4e9df34cb1baa835cebc2a99d124b481564be9ba52dd7e4d19f922f40R59